test/e2e/egress_firewall.go

```diff
@@ -290,6 +305,37 @@ spec:
 
 		}
 
+		ginkgo.By("Selecting additional IP addresses for serverNode (networking routing to secondaryIP address on other nodes is harder to achieve)")
+		// add new secondary IP from node subnet to all nodes, if the cluster is v6 add an ipv6 address
+		toCurlSecondaryNodeIPAddresses := sets.NewString()
+		// Calculate and store for AfterEach new target IP addresses.
+		var newIP string
+		if node2ndaryIPs[serverNodeInfo.name] == nil {
+			node2ndaryIPs[serverNodeInfo.name] = make(map[int]string)
+		}
+		if utilnet.IsIPv6String(e2enode.GetAddresses(&nodes.Items[1], v1.NodeInternalIP)[0]) {
+			newIP = "fc00:f853:ccd:e794::" + strconv.Itoa(2)
+			framework.Logf("Secondary nodeIP %s for node %s", serverNodeInfo.name, newIP)
+			node2ndaryIPs[serverNodeInfo.name][6] = newIP
+		} else {
+			newIP = "172.18.1." + strconv.Itoa(3)
+			framework.Logf("Secondary nodeIP %s for node %s", serverNodeInfo.name, newIP)
+			node2ndaryIPs[serverNodeInfo.name][4] = newIP
+		}
+
+		ginkgo.By("Adding additional IP addresses to node on which source pod lives")
```

Comment:
so source pod is on the same node as the server node, right?

Reply:
yes correct.